lib/eslint.js

```diff
@@ -998,6 +998,11 @@ module.exports = (function() {
             i,
             len;
 
+        // Special Node.js scope means we need to start one level deeper
+        if (scope.type === "global" && currentConfig.ecmaFeatures.globalReturn) {
```

Comment:
Can you open a separate issue about that? I'm not comfortable making such a change until I understand what's going on.